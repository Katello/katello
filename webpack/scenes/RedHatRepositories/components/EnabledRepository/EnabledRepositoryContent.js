import React from 'react';
import PropTypes from 'prop-types';
import cx from 'classnames';
import { Spinner, OverlayTrigger, Tooltip } from 'patternfly-react';
import { translate as __ } from 'foremanReact/common/I18n';

const EnabledRepositoryContent = ({ loading, disableTooltipId, disableRepository }) => (
  <Spinner loading={loading} inline>
    <OverlayTrigger
      overlay={<Tooltip id={disableTooltipId}>{__('Disable')}</Tooltip>}
      placement="bottom"
      trigger={['hover', 'focus']}
      rootClose={false}
    >
      <button
        onClick={disableRepository}
        style={{
          backgroundColor: 'initial',
          border: 'none',
          color: '#0388ce',
        }}
      >
        <i className={cx('fa-2x', 'fa fa-minus-circle')} />
      </button>
    </OverlayTrigger>
  </Spinner>
);

EnabledRepositoryContent.propTypes = {
  loading: PropTypes.bool.isRequired,
  disableTooltipId: PropTypes.string.isRequired,
  disableRepository: PropTypes.func.isRequired,
};

export default EnabledRepositoryContent;
