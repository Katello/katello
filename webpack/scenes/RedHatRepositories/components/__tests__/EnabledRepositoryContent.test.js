import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import EnabledRepositoryContent from '../EnabledRepositoryContent';

describe('Enabled Repositories Content Component', () => {
  const mockCallBack = jest.fn();

  let shallowWrapper;
  beforeEach(() => {
    shallowWrapper = shallow(<EnabledRepositoryContent
      loading
      disableTooltipId="disable-1"
      disableRepository={mockCallBack}
    />);
  });

  it('should render', async () => {
    expect(toJson(shallowWrapper)).toMatchSnapshot();
  });

  it('should run disableRepository on click', async () => {
    expect(mockCallBack).not.toHaveBeenCalled();
    shallowWrapper.find('button').at(0).simulate('click');
    expect(mockCallBack).toHaveBeenCalled();
  });
});
