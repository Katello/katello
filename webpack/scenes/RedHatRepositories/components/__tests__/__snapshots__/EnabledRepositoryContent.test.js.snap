// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Enabled Repositories Content Component should render 1`] = `
<Spinner
  className=""
  inline={true}
  inverse={false}
  loading={true}
  size="md"
>
  <OverlayTrigger
    defaultOverlayShown={false}
    overlay={
      <Tooltip
        bsClass="tooltip"
        id="disable-1"
        placement="right"
      >
        Disable
      </Tooltip>
    }
    placement="bottom"
    rootClose={false}
    trigger={
      Array [
        "hover",
        "focus",
      ]
    }
  >
    <button
      onClick={[MockFunction]}
      style={
        Object {
          "backgroundColor": "initial",
          "border": "none",
          "color": "#0388ce",
        }
      }
    >
      <i
        className="fa-2x fa fa-minus-circle"
      />
    </button>
  </OverlayTrigger>
</Spinner>
`;
